src/programs.rkt

```diff
@@ -42,8 +42,8 @@
   (cond
    [(real? x) (convert x)]
    [(bigfloat? x) (convert (bigfloat->flonum x))]
-   [(and (pair? x) (bigfloat? (car x)) (bigfloat? (cdr x)))
-      (make-rectangular (->flonum (car x)) (->flonum (cdr x)))]
+   [(and (bigcomplex? x) (bigfloat? (bigcomplex-re x)) (bigfloat? (bigcomplex-im x)))
```

Comment:
I think this should just be an invariant of bigcomplexes, right? That the `re` and the `im` are always bigfloats? Do we need to recheck?